src/server/session.ts

```diff
@@ -179,16 +178,13 @@ namespace ts.server {
         public startNew(action: (next: NextStep) => void) {
             this.complete();
             this.requestId = this.operationHost.getCurrentRequestId();
-            this.completed = false;
             this.executeAction(action);
         }
 
         private complete() {
-            if (!this.completed) {
-                if (this.requestId) {
-                    this.operationHost.sendRequestCompletedEvent(this.requestId);
-                }
-                this.completed = true;
+            if (this.requestId) {

```

Comment:
Is it possible for a request to have an id of `0`? It might be more reliable to test `this.requestId !== undefined`